src/sessions.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * @copyright Copyright (c) 2022, chandi Langecker (git@chandi.it)
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { listen } from '@nextcloud/notify_push'
+import { sessionApi } from './services/SessionApi.js'
+import store from './store/main.js'
+
+const SESSION_INTERVAL = 90 // in seconds
+
+let hasPush = false
+
+hasPush = listen('deck_board_update', (name, body) => {
+	triggerDeckReload(body.id)
+})
+
+/**
+ * is the notify_push app active and can
+ * provide us with real time updates?
+ */
+export function isNotifyPushEnabled() {
+	return hasPush
+}
+
+/**
+ * Triggers a reload of the deck, if the provided id
+ * matches the current open deck
+ *
+ * @param triggeredBoardId
+ */
+export function triggerDeckReload(triggeredBoardId) {
+	const currentBoardId = store.state.currentBoard?.id
+
+	// only handle update events for the currently open board
+	if (triggeredBoardId !== currentBoardId) return
+
+	store.dispatch('refreshBoard', currentBoardId)
```

Comment:
This is currently not enough to also update the stack/card content as the stacks are currently fetched in Board.vue.

But I think we could just combine the two calls from https://github.com/nextcloud/deck/blob/1168255f171f99c4b3873b3d1d51ec2c10d59220/src/components/board/Board.vue#L152 for that

Reply:
I had actually rather imagined extending refreshBoard and only triggering an additional reload of the stacks if necessary based on the ETag (see code below)

but for this PR I do not intend to implement actual updates for the stacks/cards, thought to limit it to the active session avatar list as the base and work on the live updates separately :)

```diff
diff --git a/src/store/main.js b/src/store/main.js
index 52c26ab7..ebe19dd9 100644
--- a/src/store/main.js
+++ b/src/store/main.js
@@ -334,10 +334,15 @@ export default new Vuex.Store({
 			commit('setAssignableUsers', board.users)
 		},
 
-		async refreshBoard({ commit }, boardId) {
+		async refreshBoard({ commit, dispatch }, boardId) {
 			const board = await apiClient.loadById(boardId)
+			const etagHasChanged = board.ETag !== this.state.currentBoard.ETag
 			commit('setCurrentBoard', board)
 			commit('setAssignableUsers', board.users)
+			if (etagHasChanged) {
+				dispatch('loadStacks', boardId)
+			}
 		},
 
 		toggleShowArchived({ commit }) {

```